src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java

```diff
@@ -85,16 +87,32 @@ public void setUp() throws Exception {
         rejectedContainer = cloudBlobClient.getContainerReference(rejectedContainerName);
     }
 
-    protected void waitForFileToBeProcessed(String fileName, String s2sToken) {
+    protected void uploadZipFile(List<String> files, String metadataFile, String destZipFilename) {
+        // valid zip file
+        testHelper.uploadZipFile(
+            inputContainer,
+            files,
+            metadataFile,
+            destZipFilename,
+            operationContext
+        );
+    }
+
+    protected EnvelopeResponse waitForEnvelopeToBeInStatus(String fileName, List<Status> awaitedStatuses) {
+        String s2sToken = testHelper.s2sSignIn(this.s2sName, this.s2sSecret, this.s2sUrl);
```

Comment:
probably this should be done only once so we can reuse. Also another thing is intermittently we do get 503 from S2S so worth doing it once ? Not strictly for this PR may be

Reply:
Good idea, but best to do it as a separate change. Currently, a test retrieve the token only once. To improve this, we'd need to introduce some sort of global cache.